list2/ex39/index.js

```diff
@@ -0,0 +1,26 @@
+import { generateRandomInteger } from "../helper.js";
+
+const ARRAY_NUMBERS = 100;
+
+export function compactArray() {
+  const completeArray = [];
+
+  for (let arrayIndex = 0; arrayIndex < ARRAY_NUMBERS; arrayIndex++) {
+    completeArray.push(generateRandomInteger());
+  }
+
+  const compactArray = [...completeArray];
+  compactArray.sort((a, b) => b - a);
+
+  while (compactArray.find((value) => value <= 0)) {
+    compactArray.pop();
+  }
```

Comment:
The use of pop() with find() in a while loop to remove values is inefficient. This approach repeatedly scans the array to find the first invalid value, then removes it from the end of the array.

You could simply have use `.filter`

```js
const compactArray = completeArray.filter(value => value !== null && value > 0);
```